### DISCUSSION TITLE
Any example to use __anext__ ?

### DISCUSSION THREAD

**Discussion by bachue** [PyO3/pyo3#2459] [Questions]
I just noticed the signature of `__anext__` is `fn __anext__(<self>) -> Option<object> or IterANextOutput`. I don't know how to use it when I have an instance of `futures::stream::TryStream`, is there any example about it ? Thanks.

**Comment by bachue** [comment]
I think there's a wrong design here, `__anext__` are forced to return `IterANextOutput`, which is an enum and actually it's the same as `IterNextOutput`. Without `Future`, I have no idea whether the Iterator should yield or return, `impl Future<Output=IterNextOutput>` would be better.

**Reply by davidhewitt** [reply]
@bachue the lack of `Future` is deliberate for now; the question of integrating runtimes between Python and Rust is complex and for the moment PyO3 is waiting for better understanding of the solution space before trying to cover this (see #1632).

With apologies I don't think we have an example for `__anext__` in the guide yet, however you can see one in our test suite here: https://github.com/PyO3/pyo3/blob/510c126f57f5b23833580857eef38bd3c2d7e6dd/tests/test_proto_methods.rs#L745
